packages/shared/lib/services/connection.service.ts

```diff
@@ -865,6 +869,8 @@ class ConnectionService {
             })
             .update({ deleted: true, credentials: {}, credentials_iv: null, credentials_tag: null, deleted_at: new Date() });
 
+        // TODO: move the following side effects to a post deletion hook
+        // so we can remove the orchestrator and logContextGetter dependencies
```

Comment:
would be nice I think to pass those side effect as a `postDeletion` function. It would simplify the dependencies. 
It is out of scope for the PR though